mapiproxy/libmapistore/backends/indexing_mysql.c

```diff
@@ -29,16 +30,302 @@
 #include "libmapi/libmapi_private.h"
 #include <mysql/mysql.h>
 #include "../../util/mysql.h"
+#include "../../util/ccan/hash/hash.h"
 #include "../../util/schema_migration.h"
 #include "mapiproxy/libmapiproxy/backends/openchangedb_mysql.h"
 
 #include <talloc.h>
-
+#include <libmemcached/memcached.h>
 
 #define MYSQL(context)	((MYSQL *)context->data)
 
 
 /**
+   \details Generate key for FMID cache
+
+   \param mem_ctx pointer to the memory context
+   \param uri pointer to the uri to use for hashing
+
+   \return String allocated with TALLOC on success, otherwise NULL
+ */
+static char *_memcached_gen_key(TALLOC_CTX *mem_ctx, const char *uri)
+{
+	return talloc_asprintf(mem_ctx, "%"PRIx64, hash64_any((void *)uri, strlen(uri), 0));
```

Comment:
I'd had used `hash64` macro because is one of the documented functions, though in this case it doesn't matter because `sizeof(*uri) == 1`.